File: app/src/main/java/kr/co/yogiyo/rookiephotoapp/diary/main/DiariesFragmentPagerAdapter.java
package kr.co.yogiyo.rookiephotoapp.diary.main;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

public class DiariesFragmentPagerAdapter extends FragmentPagerAdapter {

    private DiariesActivity context;

    public DiariesFragmentPagerAdapter(DiariesActivity context, FragmentManager fragmentManager) {
        super(fragmentManager);
        this.context = context;
    }

    @Override
    public Fragment getItem(int position) {
        return DiariesFragment.newInstance(context, position);
    }

    @Override
    public int getCount() {
        return DiariesActivity.PAGES * DiariesActivity.LOOPS;
    }
}

File: app/src/main/java/kr/co/yogiyo/rookiephotoapp/diary/main/DiariesFragmentPagerAdapter.kt
package kr.co.yogiyo.rookiephotoapp.diary.main

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter

class DiariesFragmentPagerAdapter(private val context: DiariesActivity, fragmentManager: FragmentManager)
    : FragmentPagerAdapter(fragmentManager) {

    override fun getItem(position: Int): Fragment {
        return DiariesFragment.newInstance(context, position)
    }

    override fun getCount(): Int {
        return DiariesActivity.PAGES * DiariesActivity.LOOPS
    }
}